### PR TITLE
optimize build, remove bower

### PR DESCRIPTION
**Pull request by spboyer**
- Removed bower dependency in favor of npm
- Switched out rimraf package for del see - robrich/gulp-rimraf#14
- fixed jshint issues in site.js
- added gulp build and default gulp task for clean code practices
- modified project.json to call gulp build

### CODE REVIEW

**Comment by spboyer** [comment]
@DamianEdwards - pulled in changes and resubmitted.

File: src/live.asp.net/gulpfile.js [hunk 1]
```diff
@@ -1,63 +1,103 @@
...
-gulp.task("csslint", function() {
-    return gulp.src(paths.css)
+gulp.task("csslint", ["lib"], function() {
+    return gulp.src([paths.css, "!"  + paths.minCss])
         .pipe(csslint())
         .pipe(csslint.reporter());
 });
 
-gulp.task("lint", ["jshint", "csslint"]);
+gulp.task("prepublish", ["lib", "csslint", "jshint",  "min"]);
```
Comment (DamianEdwards) [inline src/live.asp.net/gulpfile.js:101]: Why did you have to change all the dependency tasks?

File: src/live.asp.net/wwwroot/js/site.js [hunk 2]
```diff
@@ -1,10 +1,10 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved. 
+﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 (function (window, undefined) {
-    "use strict";
+    'use strict';
```
Comment (DamianEdwards) [inline src/live.asp.net/wwwroot/js/site.js:5]: Man, you really gonna force me to use `'` aren't you

File: src/live.asp.net/gulpfile.js (hunk 1 above)
Comment (spboyer) [inline src/live.asp.net/gulpfile.js:101]: a couple reasons. First, changed to ensure to return streams where appropriate and second added dependencies to each. By doing so this makes sure that the tasks run sequentially and not in parallel.

docs -> https://github.com/gulpjs/gulp/blob/master/docs/recipes/running-tasks-in-series.md

File: src/live.asp.net/wwwroot/js/site.js (hunk 2 above)
Comment (spboyer) [inline src/live.asp.net/wwwroot/js/site.js:5]: this is a jshint violation.  You can add a `.jshintrc` file and set the options. For the single quote

```
"quotmark"      : false,
```

Comment (spboyer) [inline src/live.asp.net/wwwroot/js/site.js:5]: all options here: https://github.com/jshint/jshint/blob/master/examples/.jshintrc

**Pull request merged**